### ISSUE TITLE
Integrate Agent-based control library into AixLib

### ISSUE THREAD

**Issue by FelixBue** [RWTH-EBC/AixLib#377]
## What is the problem?

- Agent-based control library isn't published anywhere

## Why do we want to solve it?

- It is ready to be published in some form and could be a suitable addition to the "Controls" package in the AixLib. The use has been shown in the domains of building energy systems and urban energy systems.

## How do we want to solve it?

- As discussed with @marcusfuchs , add the library to the AixLib with minimum working examples.

@RoozbehSangi Please let me or Marcus know about any objections and points of discussion.

**Comment by RoozbehSangi** [comment]
HVACAgentBasedControl library added.

**Comment by FelixBue** [comment]
Thanks! I will add my adjustments tomorrow.

**Comment by FelixBue** [comment]
@RoozbehSangi Did you find a way to make the library work without the additional C-Funktions in the resources folders ("Include, Library, src and test")? If not, these need to be added. 

I can do that and adjust the paths such that the resources are all in the general AixLib/Resources folder instead of AixLib/Controls/HVACAgentBasedControl/Resources. Would that be the best practice, @marcusfuchs ?

**Comment by marcusfuchs** [comment]
I would be in favor of organizing all resources in `AixLib/Resources` rather than in `AixLib/Controls/HVACAgentBasedControl/Resources`

**Comment by RoozbehSangi** [comment]
@FelixBue Those C-functions are required, but I agree with Marcus, so please change the paths to AixLib/Resources.

**Comment by PMehrfeld** [comment]
Actually they MUST be included in `AixLib/Resources` followed by a standarized folder structure.
For information and examples see chapter 12.9.4 of the current [Modelica Language Sepcifications](https://www.modelica.org/documents/ModelicaSpec33Revision1.pdf)!
Then you only need to specify the files' names without any path. The compiler will find the correct `*.a / *.h / *.dll` files according to the bit version (32 vs. 64).

**Comment by FelixBue** [comment]
Cool, thanks!

**Comment by FelixBue** [comment]
I have included all the necessary functions and the pictures in the documentation windows. I also updated the examples to use the thermalZone model of the current library. I had to make some adjustments because the IBPSA weather bus is now used in this model.

It also seems that there is only a profile for an office room available in the thermalZone model at the moment. So the examples will deliver different results now.

I do not have time to work more on that at the moment because I should start preparing for my presentation. However, everything is functional at this stage. I will make a pull request from my fork to the branch for 377 in this rep and we can see where we go from there. (Maybe I will have another day next week to work on that.)
